willa/etl/pipeline.py

```diff
@@ -42,48 +44,74 @@ def run_pipeline(vector_store: VectorStore | None = None) -> VectorStore:
     return vector_store
 
 
-def fetch_one_from_tind(tind_id: str, vector_store: VectorStore | None = None) -> VectorStore:
-    """Fetch the files for a TIND record, then load them into the VectorStore.
+def _process_one_tind_record(record: Record, vector_store: VectorStore | None = None) -> None:
+    """Process a TIND record that has been fetched.
 
-    :param str tind_id: The ID of the TIND record.
+    This allows us to have two ways into this common code: calling
+    ``fetch_one_from_tind`` allows us to fetch by TIND ID; while calling
+    ``from_from_search_query`` allows the query's matches to be used as
+    the PyMARC Record.  This saves a TIND API call to ``fetch_metadata``
+    for each record returned in the search.
+
+    :param Record record: The PyMARC Record object associated with the TIND record.
     :param vector_store: The vector store in which to store the documents.
-                         If no vector store is specified, a new in-memory
-                         vector store will be created.
-    :returns: The vector store where the processed document(s) were stored.
+                         If None, vector processing will be skipped.
     """
-    if vector_store is None:
-        vector_store = _create_vector_store()
-
-    record: Record = fetch_metadata(tind_id)
+    tind_id: str = record['001'].value()
     files: list[dict] = fetch_file_metadata(tind_id)
     file_names: list[str] = []
     docs: list[Document] = []
 
+    tind_dir = os.path.join(os.environ['DEFAULT_STORAGE_DIR'], tind_id)
+    os.mkdir(tind_dir)
+
+    metadata = pymarc_to_metadata(record)
+    with open(os.path.join(tind_dir, f"{tind_id}.json"), 'w+', encoding='utf-8') as md_file:
+        md_file.write(json.dumps(metadata))
```

Comment:
minor request: can we write out the marcxml somewhere in this process as well? iirc, the best way to do this is to call `pymarc.marcxml.record_to_xml()` with a `pymarc.Record` as its argument.[^1] the filename should be `f"{tind_id}.xml"`. 

[^1]: as opposed to instantiating `pymarc.writer.XMLWriter`, which will wrap it in an (arguably unnecessary) MARCXML `<collection>` element